classes/local/store/object_file_system.php

```diff
@@ -898,4 +913,68 @@ public function get_filesize_by_contenthash($contenthash) {
         global $DB;
         return $DB->get_field('files', 'filesize', ['contenthash' => $contenthash], IGNORE_MULTIPLE);
     }
+
+    /**
+     * Gets valid HTTP ranges for range request.
+     *
+     * @param  int          $filesize Size of the file to be served.
+     * @return object|false           Array of range
+     */
+    public function get_valid_http_ranges($filesize) {
+        $range = manager::get_header($_SERVER, 'HTTP_RANGE');
+        if (!empty($range)) {
+            preg_match('{bytes=(\d+)?-(\d+)?(,)?}i', $range, $matches);
+            if (empty($matches[3])) {
+                $ranges = new \stdClass();
+                $ranges->rangefrom = (isset($matches[1])) ? intval($matches[1]) : 0;
+                $ranges->rangeto = (isset($matches[2])) ? intval($matches[2]) : ($filesize - 1);
+                $ranges->length = $ranges->rangeto - $ranges->rangefrom + 1;
+                if ($ranges->length > 5242880) {
+                    // Stream files in 5MB-chunks.
+                    $ranges->rangeto = $ranges->rangefrom + 5242880 - 1;
+                    $ranges->length = 5242880;
+                }
+                return $ranges;
+            }
+        }
+        return false;
+    }
+
+    /**
+     * Serves range request via Pre-Signed URL.
+     *
+     * @param  string $contenthash File content hash.
+     * @param  int    $filesize    File size.
+     * @param  object $ranges      Object with rangefrom, rangeto and length properties.
+     * @return false if couldn't get data.
+     */
+    public function serve_range_request_via_presigned_url($contenthash, $filesize, $ranges) {
+        $url = $this->externalclient->generate_presigned_url($contenthash, headers_list());
+        $headers = array(
+            'HTTP/1.1 206 Partial Content',
+            'Content-Length: '.$ranges->length,
+            'Range: bytes=' . $ranges->rangefrom . '-' . $ranges->rangeto,
+        );
+
+        $curl = curl_init();
```

Comment:
please avoid using the raw php curl libs, use the moodle lib:

$curl = new curl();

This is generally preferred, but if you are already using guzzle for some reasons then its ok as long as the moodle proxy options are honored

Reply:
`new curl();` has handy features indeed. Thanks.
Done.